crates/bitwarden-vault/src/cipher/linked_id.rs

```diff
@@ -91,37 +81,9 @@ pub enum IdentityLinkedIdType {
 impl TryFrom<u32> for LinkedIdType {
     type Error = MissingFieldError;
 
-    fn try_from(value: u32) -> Result<Self, Self::Error> {
-        match value {
-            100 => Ok(LinkedIdType::Login(LoginLinkedIdType::Username)),
-            101 => Ok(LinkedIdType::Login(LoginLinkedIdType::Password)),
-            300 => Ok(LinkedIdType::Card(CardLinkedIdType::CardholderName)),
-            301 => Ok(LinkedIdType::Card(CardLinkedIdType::ExpMonth)),
-            302 => Ok(LinkedIdType::Card(CardLinkedIdType::ExpYear)),
-            303 => Ok(LinkedIdType::Card(CardLinkedIdType::Code)),
-            304 => Ok(LinkedIdType::Card(CardLinkedIdType::Brand)),
-            305 => Ok(LinkedIdType::Card(CardLinkedIdType::Number)),
-            400 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Title)),
-            401 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::MiddleName)),
-            402 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Address1)),
-            403 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Address2)),
-            404 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Address3)),
-            405 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::City)),
-            406 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::State)),
-            407 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::PostalCode)),
-            408 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Country)),
-            409 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Company)),
-            410 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Email)),
-            411 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Phone)),
-            412 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Ssn)),
-            413 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::Username)),
-            414 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::PassportNumber)),
-            415 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::LicenseNumber)),
-            416 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::FirstName)),
-            417 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::LastName)),
-            418 => Ok(LinkedIdType::Identity(IdentityLinkedIdType::FullName)),
-            _ => Err(MissingFieldError("LinkedIdType")),
-        }
+    fn try_from(val: u32) -> Result<Self, Self::Error> {
```

Comment:
The uniffi conversion was done through the deserialize impl, while this tryfrom was implemented using a hardcoded map, while I was at it I unified both of them to use the deserialize impl.